apis/v1alpha1/instrumentation_types.go

```diff
@@ -133,6 +133,18 @@ type Java struct {
 	// Resources describes the compute resource requirements.
 	// +optional
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Extensions defines java specific extensions.
+	// +optional
+	Extensions *Extensions `json:"extensions,omitempty"`
+}
+
+type Extensions struct {
+	// Image is a container image with extensions auto-instrumentation JAR.
+	Image string `json:"image"`
```

Comment:
Does the extension needs to be rebuild for every java agent version?

Reply:
I don't think it's necessary; the user just needs to package based on the javaagent version he's using.

Reply:
For some open source extensions, maybe it should provide a compatibility list for javaagent.